odoo/addons/base/models/assetsbundle.py

```diff
@@ -322,20 +361,79 @@ def save_attachment(self, type, content):
 
         return attachment
 
-    def js(self):
-        attachments = self.get_attachments('js')
-        if not attachments:
+    def js(self, is_minified=True):
+        extension = 'min.js' if is_minified else 'js'
+        if is_minified:
+            attachments = self.get_attachments(extension)
+        else:
+            attachments = self.get_attachments(extension, version_mode="NONE")
+
+        if attachments and (is_minified or self.last_modified < attachments[0].write_date):
+            return attachments[0]
+
+        if is_minified:
             content = ';\n'.join(asset.minify() for asset in self.javascripts)
-            return self.save_attachment('js', content)
-        return attachments[0]
+            return self.save_attachment(extension, content)
+        else:
+            return self.js_with_sourcemap()
+
+    def js_with_sourcemap(self):
+        """Create/modify the ir.attachment representing the not-minified content of the bundleJS
+        and the ir.attachment representing the linked sourcemap.
+        To avoid the loss of break points in the development tool when the bundleJS
+        has to be rebuilt, the bundleJS url must not change so we will not include
+        'self.version' in the ir.attachment url.
+        If any ir.attachment exists for the bundleJS, we create a new one (bundleJS and sourcemap),
+        else, we modify the 'raw' in the existing ir.attachments.
+
+        :return ir.attachment representing the un-minified content of the bundleJS
+        """
+        sourcemap_attachment = self.get_attachments('js.map', version_mode='NONE')
+        if not sourcemap_attachment:
+            sourcemap_attachment = self.save_attachment('js.map', '', with_version=False)
+        generator = SourceMapGenerator(
+            sourceRoot="../../../",
+        )
+
+        content_bundle_list = []
+        content_line_count = 0
+        line_header = 6
+        for asset in self.javascripts:
+            if asset.is_odoo_module:
+                generator.add_source(
+                    asset.url, asset._content, content_line_count, nb_skip_line_output=line_header + 3)
+                asset_line_count = len(asset.content.split("\n")) + 6
+            else:
+                asset_line_count = generator.add_source(
+                    asset.url, asset.content, content_line_count, nb_skip_line_output=line_header)
+
+            content_bundle_list.append(asset.with_header(asset.content, minimal=False))
+            content_line_count += asset_line_count
+
+        content_bundle = ';\n'.join(content_bundle_list) + f"\n//# sourceMappingURL={sourcemap_attachment.url}"
```

Comment:
f-string seems unnecessary for this one since we're doing concatenations anyway:
```suggestion
        content_bundle = ';\n'.join(content_bundle_list) + "\n//# sourceMappingURL="  + sourcemap_attachment.url
```